### PR TITLE
Fix Go modules compatibility

### PR DESCRIPTION
**Pull request by dennwc**
Currently, the package is not compatible with Go modules. If one tries to download it, the following error is returned:

```
# github.com/smacker/go-tree-sitter
../../../go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20191120151204-b4adc5db3a99/bindings.go:5:10: fatal error: vendor/tree-sitter/lib/src/lib.c: No such file or directory
 //#include "vendor/tree-sitter/lib/src/lib.c"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

This happens because Go modules tool only fetches Go _packages_, not _directories_.

All C files that are not in the same directory as Go files (thus, not inside a Go package) are pruned from the module archive.

In fact, `go build` ignores changes to C files in all sub directories and won't rebuild the package if those C files are modified! See [this thread](https://github.com/golang/go/issues/26366#issuecomment-405683150) for more context.

This PR fixes the issue by changing the vendor script to put C files directly into the corresponding Go package directories and rewriting include paths. It fixes the issue and allows to properly use Go modules with this project. 

Signed-off-by: Denys Smirnov <denis.smirnov.91@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/smacker/go-tree-sitter/17)

### CODE REVIEW

File: vendor.sh [hunk 1]
```diff
@@ -46,15 +65,15 @@ function download_typescript() {
...
-    curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/common/scanner.h" -o "typescript/common/scanner.h"
     for lang in $langs; do
-        mkdir -p "typescript/$lang/tree_sitter"
-        curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/$lang/src/tree_sitter/parser.h" -o "typescript/$lang/tree_sitter/parser.h"
+        curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/common/scanner.h" -o "typescript/$lang/scanner.h"
+        curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/$lang/src/tree_sitter/parser.h" -o "typescript/$lang/parser.h"
         for file in $files; do
             curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/$lang/src/$file" -o "typescript/$lang/$file"
-            sed -i '' -e 's/\.\.\/\.\./\.\./g' "typescript/$lang/$file"
+            sed -i 's/"\.\.\/\.\.\/common\/scanner\.h"/"scanner\.h"/g' "typescript/$lang/$file"
```
Comment (smacker) [inline vendor.sh]: this change breaks the script on macOs.

File: vendor.sh [hunk 2]
```diff
@@ -5,8 +5,27 @@
 set -e
 
 function download_sitter() {
-    rm -rf vendor
+    rm -rf vendor;
```
Comment (smacker) [inline vendor.sh]: do we need this change?

File: .gitignore [hunk 3]
```diff
@@ -1,10 +1,2 @@
...
-!/vendor/tree-sitter/lib
-/vendor/tree-sitter/lib/*
-!/vendor/tree-sitter/lib/include
-!/vendor/tree-sitter/lib/src
-!/vendor/tree-sitter/lib/utf8proc
-/vendor/tree-sitter/lib/utf8proc/*
-!/vendor/tree-sitter/lib/utf8proc/utf8proc.h
-!/vendor/tree-sitter/lib/utf8proc/*.c
+# tree-sitter library file
+lib.c
```
Comment (smacker) [inline .gitignore]: you delete this file in `vendor.sh`

**Comment by smacker** [comment]
thanks. Looks like you solution is the only right one to make go-modules work though it feels weird to put all c code into the root of the project. Let's fix compatibility with macOs and merge.

File: vendor.sh (hunk 1 above)
Comment (dennwc) [reply vendor.sh]: And the version with `-i '' -e '...'` doesn't work on Linux. Rewriting this script in Go may be way more portable. WDYT?

File: vendor.sh (hunk 2 above)
Comment (dennwc) [reply vendor.sh]: In ignores a potential error. I can remove this line as well.

File: .gitignore (hunk 3 above)
Comment (dennwc) [reply .gitignore]: Sure, but just to make sure it's not committed by accident. It will completely break the build if it does.

**Comment by smacker** [comment]
I didn't test but it should work on linux too.

File: vendor.sh (hunk 1 above)
Comment (smacker) [reply vendor.sh]: answered in another pr

**Pull request merged**